### PR TITLE
SER-1318 - Update user profile should fetch firebase info too

### PR DESCRIPTION
**Pull request by mlay1**
# **Update user profile should fetch and update with firebase info too**

### Dependencies

 - None

### Jira

* Jira - ***https://teamdocai.atlassian.net/jira/software/projects/SER/boards/73?selectedIssue=SER-1318***

### Description

**Instead of needing to make post request to both /user and /profile endpoints when updating user information on frontend, we only need to make request to /profile endpoint now. User info from firebase is now fetched and saved to db with /profile post request. Added corresponding unit test case for /profile post request.**
Not sure if I implemented this properly. I based it off of what I saw from the /feedback endpoint.

### Steps to test

***Steps to test***

### CODE REVIEW

**Comment by prashanth-docai** [comment]
@mlay1 build is failing https://console.cloud.google.com/cloud-build/builds/608851b0-b895-4a01-86dd-9879b44ab222;step=3?project=serenity-dev-b5b6

File: porter/routes/auth.js [hunk 1]
```diff
@@ -38,7 +38,15 @@ module.exports = (router) => {
 
       return profileModel
         .updateUserProfile({ dob, gender, goals }, uid)
-        .then(() => res.status(200).json({ success: true }));
+        .then( async () => {
+          res.status(200).json({ success: true });
```
Comment (JacquesLeupin) [inline porter/routes/auth.js:42]: you probably want to send a `200` in response only after the DB interaction succeeds

**Comment by mlay1** [comment]
Tested locally using simulator and with Postman; it looks to be working as intended. Test cases pass too.

File: porter/routes/auth.js (hunk 1 above)
Comment (mlay1) [reply porter/routes/auth.js]: Should be addressed with new commit

File: porter/routes/auth.spec.js [hunk 2]
```diff
@@ -185,6 +189,15 @@ describe('auth', () => {
         });
     });
 
+    it('calls userModel with the uid', () => {
```
Comment (prashanth-docai) [inline porter/routes/auth.spec.js:192]: 'calls userModel.updateUser with the uid'

Comment (mlay1) [reply porter/routes/auth.spec.js]: done

**Comment by prashanth-docai** [comment]
waiting for the build

**Pull request merged**